### ISSUE TITLE
Add "dev-dependencies" and more

### ISSUE THREAD

**Issue by baszalmstra** [prefix-dev/pixi#14]
Besides "dependencies" we should also add "dev-dependencies". These packages are also installed in the environment but they are not added to the conda package as dependencies.

One tricky thing with dev-dependencies is that they can add additional constraints on the normal dependencies. This could cause installing a lower version of a dependency during development whereas when the package is installed as a conda package we would get a higher version of a dependency. This could cause all sorts of "works-on-my-machine" issues. 

We could choose to first resolve the normal dependencies and pin them, followed by resolving the dev dependencies. If a dev-dependency would require a lower version of a package the resolution would fail. Also not ideal but it might be more transparent than the alternative.

**Comment by baszalmstra** [comment]
We talked about multiple "levels" of environments and how to support them. Our current approach is this:

- We have the standard `depencencies` (run deps), `host-dependencies`, `build-dependencies`. These are required to be able to properly build packages.
- We allow the creation of arbitrary environments on the fly through the `environment` table. Commands can specify in which environment they want to run. This allows the creation of completely standalone virtual environments. This might be useful to be able to run tools without having those tools interfere with the dependencies of the project.

Example toml:

```toml
[package]
name = "bla"

# Run deps
[dependencies] 
sdl = "*"

# host deps
[host-dependencies]
sdl_static = "*"

# build deps
[build-dependencies]
cmake = "*"

# dependencies installed for tests (which tests? how are they defined?)
[test-dependencies]

[environment.ruff]
dependencies = { ruff = "*" }

[command]
ruff = { cmd = "ruff", environment = "ruff" }
```

**Comment by baszalmstra** [comment]
Fixed by #149
